### ISSUE TITLE
Generic proxy isn't able to whitelist a host when specifying a protocol

### ISSUE THREAD

**Issue by vincentbernat** [rancher/cattle#658]
Hey!

While the code in https://github.com/rancherio/cattle/blob/master/code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/request/handler/GenericWhitelistedProxy.java seems to be written to handle the ability to proxy to HTTPS, it doesn't seem to work as expected:

```session
$ curl 'http://localhost:8080/proxy/ec2.ap-southeast-2.amazonaws.com/'
[works]
$ curl 'http://localhost:8080/proxy/http://ec2.ap-southeast-2.amazonaws.com/'
{"id":"8a77ae82-b03e-4144-b507-1b35b919c393","type":"error","links":{},"actions":{},"status":403,"code":"Forbidden","message":"Forbidden","detail":null}
$ curl 'http://localhost:8080/proxy/https://ec2.ap-southeast-2.amazonaws.com/'
{"id":"8a77ae82-b03e-4144-b507-1b35b919c393","type":"error","links":{},"actions":{},"status":403,"code":"Forbidden","message":"Forbidden","detail":null}
```

Any hint?

**Comment by vincentbernat** [comment]
I am unable to reproduce this by running cattle locally on my machine. Only the provided docker image seems to have the problem. However, the generic proxy code has never been modified. Is it possible to get the commit the Docker image was generated from?

**Comment by vincentbernat** [comment]
From `/v1/settings`, I see that the version in the Docker image is 0.58.0. I have tried with 0.58.0 and it works fine too. However, there seems to be some differences with my setup and the one in Docker. For example, here is what I get when requesting a non-whitelisted URL:

```
< HTTP/1.1 403 Forbidden
< X-API-Schemas: http://localhost:8080/v1/schemas
< X-API-ACCOUNT-ID: 1a1
< Content-Type: application/json; charset=utf-8
< Content-Length: 152
< Server: Jetty(8.1.11.v20130520)
```

In Docker, I get:

```
< HTTP/1.1 403 Forbidden
< Content-Length: 152
< Content-Type: application/json; charset=utf-8
< Server: Jetty(8.1.11.v20130520)
< X-Api-Account-Id: 1a1
< X-Api-Schemas: http://localhost:8080/v1/schemas
< Date: Wed, 08 Jul 2015 12:10:05 GMT
```

So, there is some middleware mangling the output headers. I suppose there could be one mangling the incoming URL. Still searching...

**Comment by vincentbernat** [comment]
When building the JAR with `mvn -Drelease install` and running it locally, the HTTP headers are not mangled. When copied over `/usr/share/cattle/cattle.jar` in the Docker container, the HTTP headers are mangled. Same version of Java in both cases.

**Comment by cloudnautique** [comment]
@vincentbernat the docker image for the server are generated from rancherio/rancher repo the docker file in ./server. The version of cattle is found in that file.

We tag the repo server-<VERSION> so you should be able to find the right docker file. Cattle is also tagged on the build version. So you from the dockerfile you should be able to find the corresponding rancherio/cattle tag.

**Comment by vincent99** [comment]
When run in the normal rancher/server configuration you're going through the Go websocket proxy (https://github.com/rancherio/websocket-proxy)...

@vincentbernat is there a particular reason you're looking at the proxy?

**Comment by vincentbernat** [comment]
I am trying to add a new cloud provider (Exoscale) to Rancher UI. Like for EC2, it supports security groups and therefore, I have to get the list of security groups and eventually create one. The API is available on HTTPS only.

I'll have a look at the Go websocket proxy.

**Comment by vincent99** [comment]
@wlan0 can you take a look at this; I suspect "://" is urlencoded when you get it and/or swallowed by the websocket proxy because they should have been.

It seems like we should just have a header to set the scheme instead, and default to `https://` if not provided.. I didn't actually realize the ec2 connection was cleartext.  Using the incoming scheme doesn't really make sense, a service that is http only will break when https is configured for rancher.

**Comment by vincentbernat** [comment]
vincent99: yes, I agree. I can do a PR for the HTTPS by default instead of incoming scheme if you wish.

**Comment by vincentbernat** [comment]
See PR #667. I'll open a separate bug for the issues with the Go proxy.

**Comment by vincent99** [comment]
#667 + #670 are probably good enough actually, all the current proxied hosts support SSL and I don't see any immediate need for a cleartext `http://` support.

**Comment by vincentbernat** [comment]
I propose to close this issue since this feature is not needed right now and the current proposed way to fix it is in rancher/rancher#1485.
